mysite/code/controllers/SupportedAddonsApiController.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+class SupportedAddonsApiController extends Controller
+{
+    private static $allowed_actions = [
+        'index',
+    ];
+
+    /**
+     * Set the default cache lifetime in seconds. Only used outside of "dev" environments.
+     *
+     * @config
+     * @var int
+     */
+    private static $cache_age = 86400; // 24 hours
+
+    public function index()
+    {
+        $supportedAddons = Addon::get()->filter('Supported', true)->column('Name');
+
+        $result = ['success' => true, 'addons' => $supportedAddons];
+
+        return $this->formatResponse($result);
+    }
+
+    /**
+     * Given a result payload, format as a JSON response and return
+     *
+     * @param array $data
+     * @return SS_HTTPResponse
+     */
+    protected function formatResponse(array $data)
+    {
+        $response = new SS_HTTPResponse();
+        $response
+            ->addHeader('Content-Type', 'application/json')
+            ->setBody(Convert::raw2json($data, JSON_PRETTY_PRINT));
+
+        // Don't cache anything in dev mode
+        if (Director::get_environment_type() !== 'dev') {
+            // Only cache failure messages for one minute, otherwise use the configured cache age
+            $cacheAge = empty($data['success']) ? 60 : Config::inst()->get(__CLASS__, 'cache_age');
+            $response->addHeader('Cache-Control', 'max-age=' . $cacheAge);
+        }
+
+        return $response;
+    }
```

Comment:
Do you think we should make an abstract API controller to house this method, seeing as it's re-used in both Api* controllers now?

Reply:
👍